app/components/UI/Stake/Views/InputView/StakeInputView.tsx

```diff
@@ -13,64 +13,66 @@ import Text, {
   TextColor,
   TextVariant,
 } from '../../../../../component-library/components/Texts/Text';
-import {
-  selectConversionRate,
-  selectCurrentCurrency,
-} from '../../../../../selectors/currencyRateController';
 import {
   fiatNumberToWei,
   fromTokenMinimalUnitString,
   limitToMaximumDecimalPlaces,
   renderFiat,
   renderFromTokenMinimalUnit,
-  toWei,
-  weiToFiatNumber,
 } from '../../../../../util/number';
 import Keypad from '../../../../Base/Keypad';
 import { useStyles } from '../../../../hooks/useStyles';
 import { getStakeInputNavbar } from '../../../Navbar';
 import ScreenLayout from '../../../Ramp/components/ScreenLayout';
 import CurrencyToggle from '../../components/CurrencySwitch';
 import QuickAmounts from '../../components/QuickAmounts';
-import useBalance from '../../hooks/useBalance';
 import styleSheet from './StakeInputView.styles';
 import EstimatedAnnualRewardsCard from '../../components/EstimatedAnnualRewardsCard';
 import Routes from '../../../../../constants/navigation/Routes';
+import { useStakeContext } from '../../sdk';
+import { MetaMetricsEvents, useMetrics } from '../../../../hooks/useMetrics';
 
 const StakeInputView = () => {
   const navigation = useNavigation();
   const { styles, theme } = useStyles(styleSheet, {});
-
-  const [amount, setAmount] = useState('0');
-  const [amountBN, setAmountBN] = useState<BN>(new BN(0));
-  const { balance, balanceBN, balanceFiatNumber } = useBalance();
-  const [estimatedAnnualRewards, setEstimatedAnnualRewards] = useState('-');
+  const { trackEvent } = useMetrics();
+  const {
+    balance,
+    balanceBN,
+    balanceFiatNumber,
+    sdkService,
+    amount,
+    setAmount,
+    amountBN,
+    setAmountBN,
```

Comment:
Subjective: adding the denomination to the name might make it easier to understand and work with. I say this because in this codebase the variable `balance` is sometimes formatted in different ways. I've seen `balance` formatted like

- `25000000000000000` (wei string)
- `0.025` (ETH string)
- `0.025 ETH` (formatted ETH string with suffix)

```suggestion
    amountEth,
    setAmountEth,
    amountWei,
    setAmountWei,
```

Reply:
done